src/parse/lexer.rs

```diff
@@ -653,10 +653,7 @@ where
     // Test if a digit is of a certain radix.
     fn is_digit_of_radix(c: Option<char>, radix: u32) -> bool {
         match radix {
-            2 => matches!(c, Some('0'..='1')),
-            8 => matches!(c, Some('0'..='7')),
-            10 => matches!(c, Some('0'..='9')),
-            16 => matches!(c, Some('0'..='9') | Some('a'..='f') | Some('A'..='F')),
+            2 | 8 | 10 | 16 => matches!(c, Some(c) if c.is_digit(radix)),
```

Comment:
Can we do something like this?

```rust
c.map(|c| c.is_digit(radix)).unwrap_or(false)
```

I found the matches a bit confusing with the guard

Reply:
My initial idea was:
```rust
c.filter(|c| c.is_digit(radix)).is_some()
```
it has both a `filter` and a `is_some` that indicate more readably that it's a check